src/main/scala/scalaz/nio/InetSocketAddress.scala

```diff
@@ -0,0 +1,50 @@
+package scalaz.nio
+
+import java.net.{ InetSocketAddress => JInetSocketAddress, SocketAddress => JSocketAddress }
+
+import scalaz.zio.IO
+
+abstract class SocketAddress private[nio] (private[nio] val jSocketAddress: JSocketAddress)
+
+class InetSocketAddress private[nio] (private val jInetSocketAddress: JInetSocketAddress)
+    extends SocketAddress(jInetSocketAddress) {
+
+  def port: Int = jInetSocketAddress.getPort
+
+  def hostName: String = jInetSocketAddress.getHostName
```

Comment:
From documentation `This method may trigger a name service reverse lookup` =0. I think we should wrap this thing in IO or thunk or something